WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeOrDiscountLineDetailsViewModel.swift

```diff
@@ -146,6 +151,11 @@ final class FeeOrDiscountLineDetailsViewModel: ObservableObject {
         }
         let discount = NSDecimalNumber(decimal: finalAmountDecimal)
         let priceAfterDiscount = price.subtracting(discount)
+        if priceAfterDiscount.compare(NSDecimalNumber.zero) == .orderedAscending {
+            updateDiscountDisallowedState(true)
+            return currencyFormatter.formatAmount(priceAfterDiscount) ?? ""
+        }
+        updateDiscountDisallowedState(false)
```

Comment:
Thanks for the guidance! Appreciated 💟 

I had three main issues here that made it trickier-than-expected:
1. The _-unnecessary-_ complexity of it all
2. Being forced to use a function with the price property passed in, directly from the view, for the updates.
3. I was unable to see where `updateAmount` or `updatePercentage` were called. Using both "Show Callers" and the "Call Hierarchy" showed nothing, I see now that this was because these are not directly invoked, but passed through the closure via `.onChange()`, most likely to facilitate decoupling, so we do not need to pass a viewmodel in, but this made it harder to follow its actual usage.

By removing the constraint of (2) as you mentioned, and just using the value we pass in the initializer everything became much simpler, and the tests have become also more understandable now: We pass an initial amount, we update the amount with a discount, and we assert if that's allowed or not.

Main update was done on 177128191203b9cbd92abfa18026cb158ab65cb6 , I've also updated the property name on https://github.com/woocommerce/woocommerce-ios/pull/14083/commits/9dda8ea3a57efd82d8d7eaa92290dc14d72c1947 , and made it a computed var on 6e3c39b99651f63ac385eec09a10227ae9e03e4d

I think there's still some oddities to be addressed in this feature (like how much we need a separation between `baseAmount` and `initialTotal`, or if we still need `lineType` at all) but that's for another day :D